### ISSUE TITLE
Add Close function to PooledWrokerPool and WorkerPool

### ISSUE THREAD

**Issue by cw9** [m3db/m3x#182]
It would be nice to have a Close() function for these WorkerPool interfaces, which could stop accepting new work and waits until all existing works are finished before return. 

This could make the server shutdown process a low simpler without the user having to pass in a sync.WaitGroup for each piece of work and waiting on that.

@richardartoul
